connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -154,7 +154,7 @@ public void onShutdown(String connector) {
 
     @Override
     public void onFailure(String connector, Throwable cause) {
-        statusBackingStore.putSafe(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
+        statusBackingStore.put(new ConnectorStatus(connector, ConnectorStatus.State.FAILED,
```

Comment:
This change was misguided, I've reverted that and instead made a bugfix here https://github.com/apache/kafka/pull/1745/files#r75564769